3529/2/checkin_context.rb

```diff
@@ -0,0 +1,31 @@
+require 'fileutils'
+require_relative 'path_loader'
+require_relative 'coordinates_handler'
+require_relative 'selfie_handler'
+
+module CheckinContext
+  include ImageFileWriter
+  include CoordinatesHandler
+  include SelfieHandler
+
+  def checkin!(message = nil, *)
+    save_context :checkin!
+    @path_check = PathLoader.new(payload)
+    new = message
+    if new
+      coord_handle(@path_check, 'in')
+    else
+      reply
+    end
+  end
+
+  private
+
+  def reply
+    if payload['text']
+      respond_with :message, text: 'Send me your selfie, please'
```

Comment:
wtf?

Reply:
??? 
that's my wau to hanlde this stuff (about how to communicate with user):(
I really don't know how can i make this better